pkg/build/controller/factory/factory.go

```diff
@@ -230,11 +238,20 @@ func (factory *ImageChangeControllerFactory) Create() controller.RunnableControl
 			queue,
 			cache.MetaNamespaceKeyFunc,
 			func(obj interface{}, err error, retries controller.Retry) bool {
-				kutil.HandleError(err)
+				imageStream := obj.(*imageapi.ImageStream)
 				if _, isFatal := err.(buildcontroller.ImageChangeControllerFatalError); isFatal {
+					glog.V(3).Infof("Will not retry fatal error for ImageStream update event %s/%s: %v", imageStream.Namespace, imageStream.Name, err)
+					kutil.HandleError(err)
+					return false
+				}
+				if retries.Count < maxRetries {
```

Comment:
Invert this condition - make the handleerror the block, and return true as the end of the function